packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -320,7 +320,7 @@ function preTransformRequest(server: ViteDevServer, url: string, base: string) {
   url = unwrapId(stripBase(url, base))
 
   // transform all url as non-ssr as html includes client-side assets only
-  server.transformRequest(url).catch((e) => {
+  server.transformRequest(url, { html: true }).catch((e) => {
```

Comment:
I actually don't know what the `html` option here is for. I've seen it used at

https://github.com/vitejs/vite/blob/8da04227d6f818a8ad9efc0056101968037c2e36/packages/vite/src/node/server/middlewares/transform.ts#L186

only. And that lines implies that the `url` is an `html` file? I've ran a couple apps to see when this gets triggered, but it always seems that the value is false.

Reply:
I think we should check in `indexHtml` middleware if the file is a public file before calling `transformRequest` on it. That might the simplest approach. It would be good to avoid another option as it creates a separate cacheKey in `transformRequest` and makes caching a bit less useful.

It also looks like the `html` option is used here:

https://github.com/vitejs/vite/blob/871d3533f7b29cbe4f7f8bd188362e0c9bc75d41/packages/vite/src/node/server/transformRequest.ts#L177-L181

I think this was used long long time ago where HTML is processed by `transformRequest` too, which now doesn't.

I'll open another PR with the checking approach.

Reply:
Ah I didn't saw the comments before I made mine. GitHub 🙃. I've already created https://github.com/vitejs/vite/pull/12650